honeybee/door.py

```diff
@@ -0,0 +1,334 @@
+        """A ladybug_geometry Point3D for the center of the door.
+
+        Note that this is the center of the bounding rectangle around this geometry
+        and not the area centroid.
+        """
+        return self._geometry.center
+
+    @property
+    def area(self):
+        """The area of the door."""
+        return self._geometry.area
+
+    @property
+    def perimeter(self):
+        """The perimeter of the door."""
+        return self._geometry.perimeter
+
+    @property
+    def boundary_condition(self):
+        """Boundary condition of this door."""
+        return self._boundary_condition
+
+    @boundary_condition.setter
+    def boundary_condition(self, value):
+        assert isinstance(value, (Outdoors, Surface)), \
+            'Door boundary condition must be Outdoors or Surface. ' \
+            'Got {}'.format(type(value))
+        self._boundary_condition = value
+
+    @property
+    def parent(self):
+        """Parent Face if assigned. None if not assigned."""
+        return self._parent
+
+    @property
+    def has_parent(self):
+        """Boolean noting whether this Door has a parent Face."""
+        return self._parent is not None
+
+    def set_adjacency(self, other_door):
+        """Set this door to be adjacent to another (and vice versa).
+
+        Note that this method does not verify whether the other_door geometry is
+        co-planar or compatible with this one so it is recommended that a test
+        be performed before using this method in order to verify these criteria.
+        The Face3D.is_centered_adjacent() or the Face3D.is_geometrically_equivalent()
+        methods are both suitable for this purpose.
+
+        Args:
+            other_aperture: Another Aperture object to be set adjacent to this one.
+        """
+        assert isinstance(other_door, Door), \
+            'Expected Door. Got {}.'.format(type(other_door))
+        self.boundary_condition = Surface(other_door)
+        other_door.boundary_condition = Surface(self)
+
+    def move(self, moving_vec):
+        """Move this Door along a vector.
+
+        Args:
+            moving_vec: A ladybug_geometry Vector3D with the direction and distance
+                to move the face.
+        """
+        self._geometry = self.geometry.move(moving_vec)
+
+    def rotate(self, axis, angle, origin):
+        """Rotate this Door by a certain angle around an axis and origin.
+
+        Args:
+            axis: A ladybug_geometry Vector3D axis representing the axis of rotation.
+            angle: An angle for rotation in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated.
+        """
+        self._geometry = self.geometry.rotate(axis, math.radians(angle), origin)
+
+    def rotate_xy(self, angle, origin):
+        """Rotate this Door counterclockwise in the world XY plane by a certain angle.
+
+        Args:
+            angle: An angle in degrees.
```

Comment:
I think the important thing is that each library is consistent within itself.  Except for `ladybug-geometry`, all other Ladybug Tools libraries currently use degrees as the default angle of measurement.  This is true for altitude/azimuth angles in ladybug-core and it is true of all angle measurements currently in this honyebee-core PR.  The main reasons why ladybug-geometry is the black sheep in the Ladybug Tools family is that using degrees there introduces several extra inefficiencies both because it involves an extra step for all of the internal trigonometric functions (for which there are a lot in ladybug-geometry) and because the geometry formats used by CAD interfaces are typically using radians. If you are willing to change over ladybug-core to use radians instead of degrees, we can consider a global standard for all of the libraries but I am personally in favor of keeping ladybug-geometry as the black sheep and having all other libraries use degrees.

Reply:
Sounds good to me. Where do you think we should document this?

Reply:
Good question.  Maybe we should just write it into the [global contributing Readme](https://github.com/ladybug-tools/contributing/blob/master/README.md) that all Ladybug Tools libraries are expected to use degrees except `ladybug-geometry`. Let me know if this seems like the right place and I can add it.

Reply:
Sounds good. I still think we should have it somewhere else for both users and contributors but adding it to contributing README should be a good start. Maybe we should add s note about units for each library in its README. It might get complicated now so I'm not sure.

Reply:
Ok, I added a whole section to **Contributing.MD** on [unit conventions](https://github.com/ladybug-tools/contributing/blob/master/README.md#unit-conventions), including the fact that we assume SI units by default.  "Complicated" is the word that I would choose if we wanted to have different contributing standards listed on each repo.  Let's try to have one (and preferably only one) set of contributing guidelines and only resort to a different set of standards on each repo if we have to.